### ISSUE TITLE
Reveal current file does not work with buffer opened by diffview plugin

### ISSUE THREAD

**Issue by TisnKu** [lambdalisue/fern.vim#478]
Thanks for this plugin. I recently switched from nerdtree, it is so much faster.  

Here is the issue. If I open a file via Fern explorer or Telescope search, then I can trigger `Fern . -reveal=% -drawer -toggle` to reveal current file. However, if the buffer is opened via diffview, then reveal won't work, it will always open the root directory and focus the root folder.

Steps to open a buffer via [diffview](https://github.com/sindrets/diffview.nvim/tree/766a4f210e67e522659302dc6bd8a8d3b8c08c54):
1. Setup up the plugin: use "sindrets/diffview.nvim"
2. Open a git project and edit something
3. Run `:DiffviewOpen master` to diff against master branch
4. Press `gf` to go to selected file for editing
5. In current buffer try to trigger Fern command to reveal it in directory

**Comment by lambdalisue** [comment]
What the value of `expand('%')` on the buffer opened via diffview? Fern use that value to determine what the actual file.

You can check it by

```
:echomsg expand('%')
```

On the buffer opened via diffview.

**Comment by TisnKu** [comment]
@lambdalisue It shows the file path correctly.
![image](https://github.com/lambdalisue/fern.vim/assets/10512841/5c16fb50-31dd-4036-9ffc-d3f7a4f42f13)

**Comment by TisnKu** [comment]
Nerdtree works with this situation though. Here is the link to the implementation: https://github.com/preservim/nerdtree/blob/c46e12a886b4a6618a9e834c90f6245952567115/autoload/nerdtree/ui_glue.vim#L296

**Comment by lambdalisue** [comment]
I could not reproduce the issue. Please make a minimal vimrc.

https://github.com/lambdalisue/fern.vim/assets/546312/9773371c-96b0-44c2-ae68-4bda09373f27

**Comment by TisnKu** [comment]
@lambdalisue Sorry for not clear enough about the repro steps. As long as you open the buffer before using diffview to open it, the issue cannot occur. So after git changes, simply reopen vim and use diffview to jump to that file directly.

**Comment by lambdalisue** [comment]
Could not reproduce.

https://github.com/lambdalisue/fern.vim/assets/546312/211b373d-96ba-4838-bc7c-aeab18cf8e6b

**Comment by TisnKu** [comment]
Here is a minimum init.lua for neovim. I figured out why you cannot repro. The issue exists only on windows powershell and not in wsl2 ubuntu.
```
vim.g.loaded_netrw = 1
vim.g.loaded_netrwPlugin = 1

-- Plugins
local ensure_packer = function()
  local fn = vim.fn
  local install_path = fn.stdpath("data") .. "/site/pack/packer/start/packer.nvim"
  if fn.empty(fn.glob(install_path)) > 0 then
    fn.system({ "git", "clone", "--depth", "1", "https://github.com/wbthomason/packer.nvim", install_path })
    vim.cmd([[packadd packer.nvim]])
    return true
  end
  return false
end

local packer_bootstrap = ensure_packer()

-- Plugins
require("packer").startup(function(use)
  use("wbthomason/packer.nvim")

  use("lambdalisue/fern.vim")
  use("preservim/nerdtree")
  use { 'sindrets/diffview.nvim', requires = { 'nvim-lua/plenary.nvim' },
    config = function()
      require('diffview').setup()
    end
  }
  if packer_bootstrap then
    require("packer").sync()
  end
end)
-- End plugins
```

**Comment by lambdalisue** [comment]
> As long as you open the buffer before using diffview to open it, the issue cannot occur.
>
> ---
>
>  I figured out why you cannot repro. The issue exists only on windows powershell and not in wsl2 ubuntu.

Then, could you find the difference of `expand('%')` between before and after on windows powershell?

**Comment by TisnKu** [comment]
@lambdalisue There is a difference!
Using diffview to open a buffer shows the absolute path:
![image](https://github.com/lambdalisue/fern.vim/assets/10512841/07e26dc1-0ca9-46f0-9862-e084d3ddfb22)

Directly open a buffer shows relative path:
![image](https://github.com/lambdalisue/fern.vim/assets/10512841/9389212e-3f54-4834-a07d-06275be1598f)

**Comment by TisnKu** [comment]
The same difference exists on WSL2 but that does not seem to be a problem.

**Comment by lambdalisue** [comment]
So diffview opens a path separated by slash (`/`) even on windows but the proper path should be backslash (`\`) right?

**Comment by TisnKu** [comment]
@lambdalisue Exactly. Interesting that the buffer is opened succssfully with this path and NerdTree can still locate the file.

**Comment by lambdalisue** [comment]
@TisnKu I'm sorry for late. While I could not reproduce the situation, I'm not sure if this change fixes your issue but could you try #505 ?
